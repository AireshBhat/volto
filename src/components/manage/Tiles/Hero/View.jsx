/**
 * View image tile.
 * @module components/manage/Tiles/Hero/View
 */

import React from 'react';
import PropTypes from 'prop-types';
//import { MixedTitle } from '../../../components';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <div className={['tile', 'hero'].filter(e => !!e).join(' ')}>
    <div className="product-hero">
      <img src={`${data.url}/@@images/image`} alt="" />
      <div className="product-hero-body">
        {/*<h1>
          <MixedTitle
            light={data.boldTitle}
            strong={data.title}
            breakline
            reverse
          />
        </h1>*/}
        <p>{data.description}</p>
      </div>
    </div>
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
