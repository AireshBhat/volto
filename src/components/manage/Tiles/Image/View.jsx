/**
 * View image tile.
 * @module components/manage/Tiles/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'semantic-ui-react';
import cx from 'classnames';
import { settings } from '~/config';

import { flattenToAppURL } from '../../../../helpers';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <p
    className={cx(
      'tile image align',
      {
        center: !Boolean(data.align),
      },
      data.align,
    )}
  >
    <Image
      src={
        data.url.startsWith(settings.apiPath)
          ? `${flattenToAppURL(data.url)}/@@images/image`
          : data.url
      }
      alt=""
    />
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
