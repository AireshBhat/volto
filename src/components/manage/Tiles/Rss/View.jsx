/**
 * View rss tile.
 * @module components/manage/Tiles/Rss/View
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, Feed, Image } from 'semantic-ui-react';
import { settings } from '~/config';
import Parser from 'rss-parser';

/**
 * View rss tile class.
 * @class View
 * @extends Component
 */
class View extends Component {
constructor(props){
  super(props);
  this.state = {
feed: {},
  };
}

componentDidMount() {
  (async () => {
    const parser = new Parser();
    const myfeed = await parser.parseURL(this.props.data.url);
    this.setState({
      feed: myfeed,
    });
  })()
}

render(){
return (
  <p
    className={['tile', 'image', 'align', this.props.data.align]
      .filter(e => !!e)
      .join(' ')}
  >
    {this.props.data.url.match('.rss') ? (
      <Card>
        <Feed>
          {this.state.feed.image && (
            <Feed.Event>
              <Image
                src={
                  this.state.feed.image.url
                }
                alt=""
              />
            </Feed.Event>
          )}
          <Feed.Event>
            <Feed.Date>
              {this.state.feed.lastBuildDate}
            </Feed.Date>
          </Feed.Event>
          <Feed.Event>
            {this.state.feed.feedUrl}
          </Feed.Event>
          {this.state.feed.items !== 'undefined' ? (
            <Feed.Event>
              {this.state.feed.items.forEach(item => (
                <div>
                  <Feed.Event>
                    {item.title}
                  </Feed.Event>
                  <Feed.Event>
                    {item.author}
                  </Feed.Event>
                </div>
              ))}
            </Feed.Event>
          ) : (
              null
            )}
        </Feed>
      </Card>
    ) : (
        <Container>
          <center>
            Please Enter Correct Rss Url
      </center>
        </Container>
      )}
  </p>
)
}
}

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
