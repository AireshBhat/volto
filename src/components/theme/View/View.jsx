/**
 * View container.
 * @module components/theme/View/View
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { Portal } from 'react-portal';
import { Link } from 'react-router';
import { Dropdown, Icon } from 'semantic-ui-react';
import { injectIntl, intlShape } from 'react-intl';
import { find } from 'lodash';

import {
  Comments,
  DocumentView,
  FileView,
  ImageView,
  ListingView,
  NewsItemView,
  SocialSharing,
  SummaryView,
  TabularView,
  Tags,
  Toolbar,
  Actions,
  Display,
  Types,
  Workflow,
} from '../../../components';
import { getActions, getContent } from '../../../actions';
import { getBaseUrl } from '../../../helpers';

@injectIntl
@connect(
  (state, props) => ({
    actions: state.actions.actions,
    content: state.content.data,
    pathname: props.location.pathname,
    versionId: props.location.query && props.location.query.version_id,
  }),
  {
    getActions,
    getContent,
  },
)
/**
 * View container class.
 * @class View
 * @extends Component
 */
export default class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    actions: PropTypes.shape({
      object: PropTypes.arrayOf(PropTypes.object),
      object_buttons: PropTypes.arrayOf(PropTypes.object),
      user: PropTypes.arrayOf(PropTypes.object),
    }),
    getActions: PropTypes.func.isRequired,
    /**
     * Action to get the content
     */
    getContent: PropTypes.func.isRequired,
    /**
     * Pathname of the object
     */
    pathname: PropTypes.string.isRequired,
    /**
     * Version id of the object
     */
    versionId: PropTypes.string,
    /**
     * Content of the object
     */
    content: PropTypes.shape({
      /**
       * Layout of the object
       */
      layout: PropTypes.string,
      /**
       * Allow discussion of the object
       */
      allow_discussion: PropTypes.bool,
      /**
       * Title of the object
       */
      title: PropTypes.string,
      /**
       * Description of the object
       */
      description: PropTypes.string,
      /**
       * Type of the object
       */
      '@type': PropTypes.string,
      /**
       * Subjects of the object
       */
      subjects: PropTypes.arrayOf(PropTypes.string),
      is_folderish: PropTypes.bool,
    }),
    intl: intlShape.isRequired,
  };

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    actions: null,
    content: null,
    versionId: null,
  };

  state = {
    hasObjectButtons: null,
  };

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentWillMount() {
    this.props.getActions(this.props.pathname);
    this.props.getContent(this.props.pathname, this.props.versionId);
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.pathname !== this.props.pathname) {
      this.props.getActions(nextProps.pathname);
      this.props.getContent(nextProps.pathname, this.props.versionId);
    }

    if (nextProps.actions.object_buttons) {
      const objectButtons = nextProps.actions.object_buttons;
      this.setState({
        hasObjectButtons: !!objectButtons.length,
      });
    }
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    if (!this.props.content) {
      return <span />;
    }

    let view;

    switch (this.props.content.layout) {
      case 'summary_view':
        view = <SummaryView content={this.props.content} />;
        break;
      case 'tabular_view':
        view = <TabularView content={this.props.content} />;
        break;
      case 'listing_view':
        view = <ListingView content={this.props.content} />;
        break;
      case 'news_item_view':
        view = <NewsItemView content={this.props.content} />;
        break;
      case 'file_view':
        view = <FileView content={this.props.content} />;
        break;
      case 'image_view':
        view = <ImageView content={this.props.content} />;
        break;
      default:
        view = <DocumentView content={this.props.content} />;
        break;
    }

    const viewName = view.type
      ? view.type.WrappedComponent
        ? view.type.WrappedComponent.name
        : view.type.name
      : view.constructor.name;
    const path = getBaseUrl(this.props.pathname);
    const editAction = find(this.props.actions.object, { id: 'edit' });
    const folderContentsAction = find(this.props.actions.object, {
      id: 'folderContents',
    });
    const historyAction = find(this.props.actions.object, { id: 'history' });
    const sharingAction = find(this.props.actions.object, {
      id: 'local_roles',
    });

    return (
      <div id="view">
        <Helmet
          bodyAttributes={{
            class: `view-${viewName.toLowerCase()}`,
          }}
        />
        {view}
        {this.props.content.subjects &&
          this.props.content.subjects.length > 0 && (
            <Tags tags={this.props.content.subjects} />
          )}
        <SocialSharing
          url={typeof window === 'undefined' ? '' : window.location.href}
          title={this.props.content.title}
          description={this.props.content.description || ''}
        />
        {this.props.content.allow_discussion && (
          <Comments pathname={this.props.pathname} />
        )}

        <Portal node={__CLIENT__ && document.getElementById('toolbar')}>
          <Toolbar pathname={this.props.pathname} />
        </Portal>
      </div>
    );
  }
}
