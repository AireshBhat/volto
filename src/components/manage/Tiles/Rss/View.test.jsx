import React from 'react';
import renderer from 'react-test-renderer';
import View from './View';

test('renders a view rss component', () => {
  const component = renderer.create(
    <View data={{ url: 'https://www.reddit.com/.rss' }} />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
