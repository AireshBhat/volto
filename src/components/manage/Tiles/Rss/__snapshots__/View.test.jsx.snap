// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a view rss component 1`] = `
<p
  className="tile image align"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="ui feed"
    >
      <div
        className="event"
      >
        <div
          className="date"
        />
      </div>
      <div
        className="event"
      />
    </div>
  </div>
</p>
`;
